File: Projects/PythonProject/IO_Streams/ByteStreamsExample.py
def copy_file_using_byte_stream(source_file, dest_file):
    try:
        with open(source_file, 'rb') as src, open(dest_file, 'wb') as dst:
            while True:
                byte_data = src.read(1024)  # Read in chunks for efficiency
                if not byte_data:
                    break
                dst.write(byte_data)
        print("File copied successfully!")
    except IOError as e:
        print("Error while copying file:", e)


if __name__ == "__main__":
    copy_file_using_byte_stream(
        "C:/WorkSpace/Python_Codes/8. IO Streams/text1.txt" ,
        "C:/WorkSpace/Python_Codes/8. IO Streams/destination.txt"
    )

File: Projects/JavaProject/src/com/IO_Streams/ByteStreamsExample.java
package com.IO_Streams;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class ByteStreamsExample {
    public static void copyFileUsingByteStream(String sourceFile, String destFile) {
        try (FileInputStream fis = new FileInputStream(sourceFile);
             FileOutputStream fos = new FileOutputStream(destFile)) {
            int byteData;
            while ((byteData = fis.read()) != -1) {
                fos.write(byteData);
            }
            System.out.println("File copied successfully!");
        } catch (IOException e) {
            System.out.println("Error while copying file: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        copyFileUsingByteStream("C:\\Java_Codes\\8. IO Streams\\text1.txt", "C:\\Java_Codes\\8. IO Streams\\destination.txt"); 
    }
}

